### PR TITLE
ANW-1919 port agents specs to capybara

### PR DESCRIPTION
**Pull request by thimios**
## Related JIRA Ticket or GitHub Issue
https://archivesspace.atlassian.net/browse/ANW-1919

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have authority to submit this code.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: frontend/spec/features/agents_spec.rb [hunk 1]
```diff
@@ -0,0 +1,1728 @@
...
+
+        fill_in 'agent_names__0__primary_name_', with: ''
+
+        # Click on save
+        find('button', text: 'Save Person', match: :first).click
+        element = find('.alert.alert-danger.with-hide-alert')
+        expect(element.text).to eq 'Primary Part of Name - Property is required but was missing'
+      end
+
+      it 'reports errors when updating a Person Agent with invalid data' do
```
Comment (brianzelip) [inline frontend/spec/features/agents_spec.rb:528]: This spec should be labeled 'can add a related agent', not the copied-from-above 'reports errors...'.

File: frontend/spec/features/agents_spec.rb [hunk 2]
```diff
@@ -0,0 +1,1728 @@
...
+    it 'hides agent_conventions_declarations from form' do
+      expect(page).to_not have_css '#agent_person_agent_conventions_declaration'
+    end
+
+    it 'hides agent_maintenance_histories from form' do
+      expect(page).to_not have_css '#agent_person_agent_maintenance_history'
+    end
+
+    it 'hides agent_other_agency_codes from form' do
+      expect(page).to_not have_css '#agent_person_agent_sources'
```
Comment (brianzelip) [inline frontend/spec/features/agents_spec.rb:1645]: The selector here should read `'#agent_person_agent_other_agency_codes'`.

File: frontend/spec/features/agents_spec.rb [hunk 3]
```diff
@@ -0,0 +1,1728 @@
...
+        select 'Not Specified', from: 'agent_agent_genders__0__gender_'
+
+        # Click on save
+        find('button', text: 'Save Person', match: :first).click
+        element = find('.alert.alert-success.with-hide-alert')
+        expect(element.text).to eq "Agent Created"
+
+        expect(find('#agent_agent_genders__0__gender_').value).to eq 'not_specified'
+      end
+
```
Comment (brianzelip) [inline frontend/spec/features/agents_spec.rb:944]: The `'can add date to a gender'` spec was not included here, see https://github.com/archivesspace/archivesspace/blob/v3.5.0/frontend/spec/selenium/spec/agents_spec.rb#L584

**Review by brianzelip — CHANGES_REQUESTED** [review]
@thimios Nice work! There are a few points of follow up...

File: frontend/spec/features/agents_spec.rb (hunk 2 above)
Comment (thimios) [reply frontend/spec/features/agents_spec.rb:1645]: This is actually a duplicate example that I will remove. But you are right there that it needs to check that no '#agent_person_agent_other_agency_codes' are there, which is actually already done three examples above. @brianzelip

**Review by brianzelip — APPROVED** [review]
👍🏼

**Pull request merged**